src/runtime_src/core/pcie/linux/system_linux.cpp

```diff
@@ -179,20 +179,24 @@ get_device_id(const std::string& bdf) const
     return system::get_device_id(bdf);
 
   unsigned int i = 0;
-  for (auto dev = get_pcidev(0); dev; dev = get_pcidev(++i)) {
+  try {
+    while (1) {
+      auto dev = get_pcidev(i);
```

Comment:
Hmm,  I get the try/catch, but not the `while (1)`,   Seems like `i` is never incremented so this would be an infinite loop?   Isn't the for loop better?